mcs/class/System.Net.Http/CFNetworkHandler.cs

```diff
@@ -173,6 +195,43 @@ CFHTTPMessage CreateWebRequestAsync (HttpRequestMessage request)
 		}
 
 		protected internal override async Task<HttpResponseMessage> SendAsync (HttpRequestMessage request, CancellationToken cancellationToken)
+		{
+			return await SendAsync (request, cancellationToken, true).ConfigureAwait (false);
+		}
+		
+		static async Task<HttpRequestMessage> CreateHttpRequestMessageRedirectAsync (HttpRequestMessage req)
+		{
+			var str = req.RequestUri.OriginalString;
+			var clone = new HttpRequestMessage (req.Method, str);
+
+			// Copy the request's content (via a MemoryStream) into the cloned object
+			var ms = new MemoryStream ();
+			if (req.Content != null) {
+				await req.Content.CopyToAsync (ms).ConfigureAwait (false);
```

Comment:
This does not look right. What if the content was already uploaded, why are we uploading it again?